### PR TITLE
[API] [ExchangeRate] Add currency filter

### PR DESCRIPTION
**Pull request by Tomanhez**
| Q               | A
| --------------- | -----
| Branch?         | api
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

This PR based on https://github.com/Sylius/Sylius/pull/11178

Screenshot 2020-03-10 at 11 15 10

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/filters.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<!--
+ This file is part of the Sylius package.
+ (c) Paweł Jędrzejewski
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service id="Sylius\Bundle\ApiBundle\Filters\ExchangeRateFilter">
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/filters.xml]:
```suggestion
        <service id=""sylius.api.filter.exchange_rate class="Sylius\Bundle\ApiBundle\Filters\ExchangeRateFilter">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/filters.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            <tag name="api_platform.filter" id="exchange_rate_filter"/>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/filters.xml]: Have you check convention in ApiPlatform itself?

Comment (Tomanhez) [reply src/Sylius/Bundle/ApiBundle/Resources/config/filters.xml]: Yes, its not working without tag name and tag id, Configuration in ApiPlatform docs is available only for yaml and annotation, I not saw tag id there, but I investigate that its not working without

File: src/Sylius/Behat/Client/ApiClientInterface.php [hunk 3]
```diff
@@ -29,7 +29,9 @@ public function buildCreateRequest(string $resource): void;
 
     public function buildUpdateRequest(string $resource, string $id): void;
 
-    /** @param string|int $value */
+    public function buildFilter(array $filters): void;
+
+    /** @param string|int */
```
Comment (GSadee) [inline src/Sylius/Behat/Client/ApiClientInterface.php]:
```suggestion
    /** @param string|int $value */
```

File: src/Sylius/Behat/Client/ApiPlatformClient.php [hunk 4]
```diff
@@ -73,7 +76,12 @@ public function buildUpdateRequest(string $resource, string $id): void
         $this->request['body'] = json_decode($this->client->getResponse()->getContent(), true);
     }
 
-    /** @param string|int $value */
+    public function buildFilter(array $filters): void
+    {
+        $this->filters = $filters;
+    }
+
+    /** @param string|int */
```
Comment (GSadee) [inline src/Sylius/Behat/Client/ApiPlatformClient.php]:
```suggestion
    /** @param string|int $value */
```

**Comment by lchrusciel** [comment]
Thank you, Tomasz! :tada:

**Comment by lchrusciel** [comment]
Part of #11250

**Pull request merged**